addons/mail/static/src/core/common/im_status.xml

```diff
@@ -4,14 +4,14 @@
     <t t-name="mail.ImStatus">
         <div class="o-mail-ImStatus d-flex justify-content-center flex-shrink-0 align-items-center rounded-circle bg-inherit" t-att-class="props.className" t-att-style="props.style">
             <span class="d-flex flex-column" name="icon">
-                <t t-if="!props.thread or !props.thread.hasOtherMembersTyping">
-                    <i t-if="props.persona.im_status === 'online'" class="fa fa-circle text-success" title="Online" role="img" aria-label="User is online"/>
-                    <i t-elif="props.persona.im_status === 'away'" class="fa fa-circle o-away" title="Idle" role="img" aria-label="User is idle"/>
-                    <i t-elif="props.persona.im_status === 'offline'" class="fa fa-circle-o text-700" title="Offline" role="img" aria-label="User is offline"/>
-                    <i t-elif="props.persona.im_status === 'bot'" class="fa fa-heart text-success" title="Bot" role="img" aria-label="User is a bot"/>
+                <t t-if="(!props.member or !props.member.isTyping) and persona">
+                    <i t-if="persona.im_status === 'online'" class="fa fa-circle text-success" title="Online" role="img" aria-label="User is online"/>
+                    <i t-elif="persona.im_status === 'away'" class="fa fa-circle o-away" title="Idle" role="img" aria-label="User is idle"/>
+                    <i t-elif="persona.im_status === 'offline'" class="fa fa-circle-o text-700" title="Offline" role="img" aria-label="User is offline"/>
+                    <i t-elif="persona.im_status === 'bot'" class="fa fa-heart text-success" title="Bot" role="img" aria-label="User is a bot"/>
                     <i t-else="" class="fa fa-fw fa-question-circle" title="No IM status available"/>
                 </t>
-                <Typing t-if="props.thread" channel="props.thread" size="'medium'" displayText="false" />
+                <Typing t-if="props.member?.isTyping" member="props.member" channel="props.member.threadAsTyping" size="'medium'" displayText="false" />
```

Comment:
`channel` in `Typing` can also be optional, as it can be deduced from `props.member.thread`